lib/game.rb

```diff
@@ -0,0 +1,73 @@
+require 'pry'
+class Game
+  attr_accessor :board
+  attr_accessor :player_1, :player_2
+
+  WIN_COMBINATIONS = [
+    [0,1,2],  #1
+    [3,4,5], #2
+    [6,7,8], #3
+    [0,3,6], #4
+    [0,4,8],  #5
+    [1,4,7],  #6
+    [2,5,8],  #7
+    [2,4,6]  #8 / 8 win combos
+  ]
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @board = board
+    @player_1 = player_1
+    @player_2 = player_2
+  end
+
+  def current_player
+    board.turn_count%2 == 0 ? @player_1 : @player_2
+  end
+
+  def over?
+    board.full? #|| won?
+  end
+
+  def won?
+    win_combo = false #default if there are no win_combos
+    # board.cells[x] compares to win_combos , and if win_combos[z] all == "x", then return the combo
+    WIN_COMBINATIONS.each do |numarray|
```

Comment:
Here you can use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect) to return the winning combination or `nil` if none is found.

```ruby
  def won?
    WIN_COMBINATIONS.detect do |combo|
      @board.cells[combo[0]] == @board.cells[combo[1]] &&
      @board.cells[combo[0]] == @board.cells[combo[2]] &&
      @board.taken?(combo[0] + 1)
    end
  end
```